tests/Feature/Router/ErrorHandlingTest.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GacelaTest\Feature\Router;
+
+use Gacela\Router\Entities\Request;
+use Gacela\Router\Router;
+use GacelaTest\Feature\HeaderTestCase;
+
+final class ErrorHandlingTest extends HeaderTestCase
+{
+    public function test_not_redirect_non_registered_method(): void
```

Comment:
I think this name could be improved, it is not correctly defining the situation under test, it should be test_respod_status_404_when_the_uri_does_not_matches.